feat: publishing mail attachment too big events upon mail backup

karolsojko opened the pull request:
If the mail attachment size is bigger than the allowed size (configurable via env vars) then this will publish an SNS event instead of sending the email. The event will be picked up by the `email` service which will send the appropriate message to the user.

mobitar commented:
How would users unsubscribe from this message? We'd need a mute mechanism otherwise they'd get this email every day until they uninstall the extension. Would could either have a mute link, or a link to uninstall the extension. The latter would require a mostly new API.

karolsojko commented:
@mobitar applied changes:
- added `allowedSize` `attachmentSize` and `settingsId` to the event payload that email service will pick up
- the event will not be published if the extension setting has `mute_emails` enabled
- removed the `perform_email_backup` from `user` as it is not used anywhere (causes additional maintenance with `ArchiveMailer.data_backup` method)
